src/main/java/com/mappy/fpm/batches/tomtom/dbf/timedomains/TimeDomainsProvider.java

```diff
@@ -1,35 +1,46 @@
 package com.mappy.fpm.batches.tomtom.dbf.timedomains;
 
-import com.google.common.collect.Multimap;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
 import com.google.common.collect.TreeMultimap;
 import com.mappy.fpm.batches.tomtom.TomtomFolder;
 import com.mappy.fpm.batches.tomtom.dbf.TomtomDbfReader;
 import lombok.extern.slf4j.Slf4j;
+
 import org.jamel.dbf.structure.DbfRow;
 
 import javax.inject.Inject;
 import javax.inject.Singleton;
 import java.util.Collection;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
 
 @Slf4j
 public class TimeDomainsProvider extends TomtomDbfReader {
 
-    private final Multimap<Long, TimeDomains> timeDomainsMap = TreeMultimap.create();
+    private final Map<Long, ArrayListMultimap<Integer, TimeDomains>> timeDomainsMap = Maps.newHashMap();
```

Comment:
We now index time domains by section id AND sequence number (a restriction primary key is composed of the section id, and a sequence number)